app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingLinksRepository.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>.
+ */
+
+package com.keylesspalace.tusky.components.trending
+
+import com.keylesspalace.tusky.network.MastodonApi
+import javax.inject.Inject
+
+class TrendingLinksRepository @Inject constructor(
```

Comment:
This looks small? :-)

What is the future (extension) of this?

Reply:
Yes, but it serves an important role -- it decouples the view model from the API.

The goal is:

- Fragments/Activities -- display the UI state, convert user actions in to actions to send to the view model
- ViewModel -- fetches data from repositories, converts it to UI state for display. Turns user actions from the UI into operations the repository should perform on the data
- Repository -- manages access to the data. Shields the ViewModel from how that data is stored / where it is fetched from
- PagingSource / RemoteMediator -- fetches the underlying data

https://developer.android.com/topic/architecture is the place to read about all this.

We're currently in the middle of transitioning the existing code to this. New code should follow this model.